lib/registry/registry.rb

```diff
@@ -17,7 +17,9 @@ class Registry # rubocop:disable Metrics/ClassLength
     # @option options [Hash] :http_options Extra options for RestClient::Request.execute.
     def initialize(uri, options = {})
       @uri = URI.parse(uri)
-      @base_uri = "#{@uri.scheme}://#{@uri.host}:#{@uri.port}#{@uri.path}"
+      @base_uri = +"#{@uri.scheme}://#{@uri.host}:#{@uri.port}#{@uri.path}"
+      # `URI.join("https://example.com/foo/bar", "v2")` drops `bar` in the base URL. A trailing slash prevents that.
+      @base_uri << '/' unless @base_uri.end_with? '/'
```

Comment:
Indeed.

Reply:
To be clearer, when an URL does not end with a trailing slash, the last part is assumed to be a file, so joining another path to it will make the path relative to the directory and not the file.

To give a more natural example, /foo/index.html joined with style.css would become /foo/style.css, white foo/index/ joined with style.css becomes /foo/index/style.css.